### DISCUSSION TITLE
QUESTION: Uninstall modules not found after or before migrate?

### DISCUSSION THREAD

**Discussion by jmmicha12** [OCA/OpenUpgrade#2776] [Q&A] [answered]
Hi, I want to migrate from V8 to V11, but during the migration from V8 to V11 there are modules from previous versions that when migrating are no longer there. 
Should I uninstall the modules in version 9 that were from version 8 but are gone before migrating to version 10 or do I continue the migration process to version 10 without uninstalling anything? Same question for version 10 to version 11.

Thank you very much.

**Comment by pedrobaeza** [comment]
Converted to question in the new Discussions zone.

**Comment by jmmicha12** [comment]
Sorry, I didn't know about the discussion section.

**Comment by legalsylvain** [comment, marked as answer]
Hi. 
- if you want to keep the module, and if your module exist in V8 and V11 but not in V9 and V10, you have nothing to do. 
- if you don't want to keep the module, if your module doesn't contain data or if you don't care about the data, uninstall the module as soon as possible. it will make the migration easier and faster.
- if you don't want to keep the module, but if your module contains data that you want to preserve, you have to migrate to 11, then use a new module to recover the data, and then uninstall the old module.
